### PR TITLE
fix: check null locations

### PR DESCRIPTION
**Pull request by srchase**
Fixes: https://github.com/aws/aws-sdk-js-v3/issues/871

Need to check that locations are not null before attempting to deserialize.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by srchase** [comment]
With this fix, responses are deserialized like this:
```
{
  '$metadata': {
    httpStatusCode: 200,
    httpHeaders: {
      'content-type': 'application/json',
      date: 'Thu, 06 Feb 2020 17:26:11 GMT',
      'x-amzn-requestid': '...',
      'content-length': '372',
      connection: 'keep-alive'
    },
    requestId: '...',
    retries: 0,
    totalRetryDelay: 0
  },
  __type: 'PostTextResponse',
  dialogState: 'ElicitSlot',
  intentName: 'BookHotel',
  message: 'What city will you be staying in?',
  messageFormat: 'PlainText',
  responseCard: undefined,
  sentimentResponse: undefined,
  sessionAttributes: undefined,
  sessionId: '2020-02-06T17:22:12.905Z-LzBxvqUs',
  slotToElicit: 'Location',
  slots: { CheckInDate: null, Location: null, Nights: null, RoomType: null }
```

Does it make sense to mix `null` and `undefined`?

**Comment by kstich** [comment]
Do the non-structure member targets need this treatment as well?

From the above:
```
 slots: { CheckInDate: null, Location: null, Nights: null, RoomType: null }
```

**Comment by trivikr** [comment]
> Does it make sense to mix `null` and `undefined`?

Yes, it makes sense to add both `null` and `undefined` checks before deserializing.

TLDR version:
* The value null represents the intentional absence of any object value ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null))
* undefined is a primitive value automatically assigned to variables that have just been declared, or to formal arguments for which there are no actual arguments. ([MDN](https://developer.mozilla.org/en-US/docs/Glossary/Undefined))
* null is an empty or non-existent value, where as undefined typically means a variable has been declared, but not defined. Both null and undefined are primitive values, as well as falsy.

**Comment by srchase** [comment]
```
sentimentResponse: undefined,
sessionAttributes: undefined,
...
slots: { CheckInDate: null, Location: null, Nights: null, RoomType: null }
```

The service explicitly returns `null` values for the `slots` string to string map.  This change results in the correct behavior.
